exchange/seat_non_bids.go

```diff
@@ -0,0 +1,68 @@
+package exchange
+
+import (
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/exchange/entities"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type seatNonBids struct {
+	seatNonBidsMap map[string][]openrtb_ext.NonBid
+}
+
+func newSeatNonBids() seatNonBids {
+	return seatNonBids{
+		seatNonBidsMap: make(map[string][]openrtb_ext.NonBid),
+	}
+}
+
+func (snb *seatNonBids) add(bid *entities.PbsOrtbBid, nonBidReason int, seat string) {
+	if bid == nil || bid.Bid == nil {
+		return
+	}
+	if snb.seatNonBidsMap == nil {
+		snb.seatNonBidsMap = make(map[string][]openrtb_ext.NonBid)
+	}
+	nonBid := openrtb_ext.NonBid{
+		ImpId:      bid.Bid.ImpID,
+		StatusCode: nonBidReason, //
+		Ext: openrtb_ext.NonBidExt{
+			Prebid: openrtb_ext.Prebid{Bid: openrtb_ext.Bid{
+				Bid: openrtb2.Bid{
+					Price:   bid.Bid.Price,
+					ADomain: bid.Bid.ADomain,
+					CatTax:  bid.Bid.CatTax,
+					Cat:     bid.Bid.Cat,
+					DealID:  bid.Bid.DealID,
+					W:       bid.Bid.W,
+					H:       bid.Bid.H,
+					Dur:     bid.Bid.Dur,
+					MType:   bid.Bid.MType,
+				},
+				OriginalBidCPM: bid.OriginalBidCPM,
+				OriginalBidCur: bid.OriginalBidCur,
+			}},
+		},
+	}
+
+	nonBids := snb.seatNonBidsMap[seat]
+
+	if nonBids == nil {
+		snb.seatNonBidsMap[seat] = []openrtb_ext.NonBid{nonBid}
+	} else {
+		snb.seatNonBidsMap[seat] = append(nonBids, nonBid)
+	}
```

Comment:
This condition isn't necessary. The `append` method will gracefully handle nil slices. This could just be `snb.seatNonBidsMap[seat] = append(nonBids, nonBid)`

Reply:
Done